src/test/kotlin/snyk/advisor/api/AdvisorApiClientTest.kt

```diff
@@ -163,10 +165,43 @@ class AdvisorApiClientTest {
         assertThat(nonExistingPackageInfo?.name, equalTo("non-existing-package"))
         assertThat(nonExistingPackageInfo?.error, equalTo("not found"))
     }
+
+    @Test
+    fun `getPackagesInfo should return correct List of PackageInfo`() {
+        val expectedJsonFile = "scores-npm-packages_multiple-packages-payload_200.json"
+
+        server.enqueueResponse(expectedJsonFile, 200)
+
+        val expectedInfos = Gson().fromJson(
+            javaClass.classLoader.getResource("$FIXTURES_BASE_DIR/$expectedJsonFile")!!.readText(),
+            Array<PackageInfo>::class.java
+        ).toList()
+
+        val expectedNames = expectedInfos.map { it.name }
+
+        assertInfos(
+            expectedInfos,
+            clientUnderTest.getPackagesInfo(AdvisorPackageManager.NPM, expectedNames)
+        )
+    }
+
+    private fun assertInfos(expectedInfos: List<PackageInfo>, actualInfos: List<PackageInfo>?) {
+        assertThat(actualInfos, notNullValue())
+        assertThat(actualInfos, hasSize(expectedInfos.size))
+
+        actualInfos?.forEachIndexed { index, actual ->
+            val expected = expectedInfos[index]
+            assertThat(actual.name, equalTo(expected.name))
+            assertThat(actual.score, equalTo(expected.score))
+            assertThat(actual.pending, equalTo(expected.pending))
+            assertThat(actual.error, equalTo(expected.error))
+            assertThat(actual.labels, equalTo(expected.labels))
```

Comment:
assert DTO instead of each field? if unit test will fail you will see more meaningful messages